lessons/Lesson-3-Ruby-Part-1.md

```diff
@@ -9,77 +9,96 @@ You should fork [this repository](https://github.com/Code-the-Dream-School/ROR-l
 
-- puts and print output strings to the console.  There is no console.log.  The puts function puts a string with a line end, and the print function puts a string without the line end.
-- gets gets a line of input from the console and returns a string.  Of course, this is rarely done in JavaScript.
-- gets.chomp gets a line of input from the console and returns a string, having removed the line end.
-- Comments start with a # in Ruby.
-- For string interpolation, you used the back-tic in JavaScript, and then ```${}```.  For string interpolation in Ruby, you use the double quotes, and then ```#{}```.
+That's it.  There is no `const` or `let` or `var`.  Variables should be named in snake case, which means that if you have several words in your variable name, you separate them with an underscore.  It's good, of course, to use descriptive variable names.  A couple of other new things:
 
-Ok, time for your first program.  Create first.rb in this repository, as follows:
-```
+- `puts` and `print` output strings to the console.  There is no `console.log`.  The `puts` method puts a string with a line end, and the `print` method puts a string without the line end.
+- `gets` gets a line of input from the console and returns a string.  Of course, this is rarely done in JavaScript.
+- `gets.chomp` gets a line of input from the console and returns a string, having removed the line end.
+- Comments start with a `#` in Ruby.
+- For string interpolation, you used the backtick in JavaScript, and then `${}`.  For string interpolation in Ruby, you use the double quotes, and then `#{}`.
+
+Ok, time for your first program.  Create `first.rb` in this repository, as follows:
+
+```ruby
 puts "Enter your name." # Here we put a prompt to the console
 name = gets.chomp       # Here we collect the answer
 puts "Well hello, #{name}!"  # We respond with a greeting
 value_a = value_b
 ```
+
 Now run it in your terminal:
-```
-ruby first.rb
-```  
-You'll get an error (after putting in your name) because the last line references an uninitialized variable. Note that Ruby tells you where the error is.  So remove that last line and run the program again.
 
-The puts call is a call to a function.  Ruby does not require parentheses for function calls, provided that the argument is the last thing on the line.  You can put the parentheses in, and sometimes it is wise to do so for clarity.  The gets call is also a call to a function, and that call is chained to another function call, chomp.  So you could write this as
+```shell
+ruby first.rb
 ```
+
+You'll get an error (after putting in your name) because the last line references an undefined variable. Note that Ruby tells you where the error is.  So remove that last line and run the program again.
```

Comment:
I replaced "uninitialized" with "undefined" because the latter makes more sense for Ruby variables. While in other languages, you can declare variables without initializing them, in Ruby they're simply defined or undefined.